### ISSUE TITLE
g3dj to g3db converter as executable

### ISSUE THREAD

**Issue by Duna** [Dancovich/libgdx_g3db_converter#1]
We would like to have an executable or an applet for easier conversion of g3dj to g3db format

**Comment by Dancovich** [comment]
Calling the compiled class with the 'java' executable as seen in the README file can serve as an executable, you can use the command in scripts and such.

Is there a specific scenario where you believe this wouldn't work?

**Comment by Duna** [comment]
It has libgdx dependency, it is not pure java

**Comment by Duna** [comment]
It has libgdx dependency, it is not pure java. I didn't succed to call it

**Comment by Dancovich** [comment]
Yes, this is just a utility class so it doesn't have standalone code to do the conversion.

To run it with the libGDX dependency download the jar from the libGDX page, place it on the root of the classpath (where the "br" folder is) and run `java -cp libgdx.jar:. br.com.questingsoftware.libgdx.g3db.G3DBConverter /path/to/models/folder`. This will add the entire libgdx.jar library on the classpath so my class can find the required dependencies.

Creating a standalone executable could theoretically be done by using something like Launch4j to create an embedded executable that would automatically run the above command for you. It would still require LibGDX but I believe the Launch4J will actually pack both my class and the LibGDX jar in a single executable file. You could give it a go.

The current version of my exporter script for Blender can export directly to G3DB now. I created this little utility just so that I could call it from my game code and still have the G3DJ files to quickly check if my 3D models had any problems. The script is here: https://github.com/Dancovich/libgdx_blender_g3d_exporter

**Comment by Duna** [comment]
Does the plug-in exports the Ids of textures too, or just the mesh and animations?

**Comment by Dancovich** [comment]
It also exports ids for textures. It won't copy the texture to where you save the G3D file though, you have to do it yourself. It will also export the path to the texture relative to where the BLEND file is, so what I do is save the BLEND and G3D file on the same place and have a "tex" subfolder with all the textures, that way the exported texture will have a path like "./tex/texture.png".

Keep in mind the Blender exporter is limited to the capabilities of the G3D format. It will only export textures UV mapped and doesn't support procedural materials. It also only supports quaternion rotations so don't use Euler rotations when positioning and animating meshes.

In general if you just download a 3D model in Blender and try to export it most of the time the model will use a feature the G3D format doesn't support and you won't get the same model on your game. Keep the limitations of the exporter and G3D format in mind when creating your models.
